pkg/server/image.go

```diff
@@ -337,19 +337,58 @@ func (ih *ImageHandler) scaleImage(ctx context.Context, fileRef blob.Ref) (*form
 var singleResize singleflight.Group
 
 func (ih *ImageHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request, file blob.Ref) {
-	ctx := req.Context()
 	if !httputil.IsGet(req) {
 		http.Error(rw, "Invalid method", http.StatusBadRequest)
 		return
 	}
+
+	ctx := req.Context()
+	res, err := ih.Search.Describe(ctx, &search.DescribeRequest{
+		BlobRef: file,
+		Depth:   1,
+	})
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	db, ok := res.Meta[file.String()]
+	if !ok {
+		http.Error(rw, "blob meta not found", http.StatusInternalServerError)
+		return
+	}
+
+	if db.File.MIMEType == "image/svg+xml" {
+		_, etag, _ := strings.Cut(file.String(), "-")
+		fr, err := schema.NewFileReader(ctx, ih.Fetcher, file)
+		if err != nil {
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		defer fr.Close()
+
+		h := rw.Header()
+		if !disableThumbCache {
+			h.Set("Expires", time.Now().UTC().Add(oneYear).Format(http.TimeFormat))
+			h.Set("Etag", strconv.Quote(etag))
+		}
+		h.Set("Content-Type", db.File.MIMEType)
+
+		if req.Method == "GET" {
```

Comment:
I don't think you need (or want) this. For a HEAD request, you still want to set the Content-Length, etc. ServeContent will do that for you.